Leanplum-SDK/Classes/LPDatabase.m

```diff
@@ -0,0 +1,185 @@
+//
+//  LPDatabase.m
+//  Leanplum
+//
+//  Created by Alexis Oyama on 6/9/17.
+//  Copyright (c) 2017 Leanplum, Inc. All rights reserved.
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+
+#import "LPDatabase.h"
+#import "LPFileManager.h"
+#import "LeanplumInternal.h"
+#import "Constants.h"
+#import <sqlite3.h>
+
+@implementation LPDatabase
+
+/**
+ * Create/Open SQLite database.
+ */
+- (id)init
+{
+    if (self = [super init]) {
+        const char *sqliteFilePath = [[LPDatabase sqliteFilePath] UTF8String];
+        sqlite3 *sqlite;
+        int result = sqlite3_open(sqliteFilePath, &sqlite);
+        if (result != SQLITE_OK) {
+            LPLog(LPError, @"Fail to open SQLite with result of %d", result);
+            return self;
+        }
+        
+        // Create tables.
+        [self runQuery:@"CREATE TABLE IF NOT EXISTS event ("
```

Comment:
I don't think this needs to be in a constant. It is only used by the Database and the EventDataManager. It will be harder to read if the query is @"INSERT INTO %@ (%@) " with parameters.

Also column have types which will make it quite awkward and again hard to read.
"CREATE TABLE IF NOT EXISTS %@ (
 %@ TEXT NOT NULL)", LP_EVENT_TABLE, LP_DATA_COLUMN